src/main/java/org/zone/region/flag/entity/player/move/preventing/PreventPlayersListener.java

```diff
@@ -53,4 +58,142 @@ public void onPlayerMove(MoveEntityEvent event, @Getter("entity") Player player)
         event.setCancelled(true);
     }
 
+    @Listener
+    public void onPlayerLoginEvent(ServerSideConnectionEvent.Join joinevent) {
+        ServerPlayer serverPlayer = joinevent.player();
+        Optional<Zone> opZone = ZonePlugin
+                .getZonesPlugin()
+                .getZoneManager()
+                .getPriorityZone(joinevent.player().world(), joinevent.player().position());
+
+        if (opZone.isEmpty()) {
+            return;
+        }
+
+        if (!(opZone.get().containsFlag(FlagTypes.PREVENT_PLAYERS))) {
+            return;
```

Comment:
This is checked in the line ``flag.isEmpty`` so you can remove this if statement completely

Reply:
k